apis/vpcresources/v1alpha1/cninode_types.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FeatureName is a type of feature name supported by EKS networking. It can be Security Group for Pods, custom networking, or others
+type FeatureName string
+
+const (
+	SecurityGroupsForPods FeatureName = "SecurityGroupsForPods"
+	CustomNetworking      FeatureName = "CustomNetworking"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+// CNINodeSpec defines the desired state of CNINode
+type CNINodeSpec struct {
+	Features []FeatureName `json:"features,omitempty"`
+}
+
+type VPCResource string
+
+// +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Features",type=string,JSONPath=`.spec.features`,description="The features delegated to VPC resource controller"
+// +kubebuilder:resource:shortName=cnd
```

Comment:
```suggestion
// +kubebuilder:resource:shortName=cni
```

Reply:
Yeah, thought about `cni` as well, but it is definitely a conflict with well known `CNI`. Waiting for comments.

Reply:
maybe `awsvpccni` or just `vpccni`? Btw, was there any other options instead of CNINode or CNINodes? Also this is specific to `aws-vpc-cni` only and not `amazon-vpc-cni-plugins` right? Is there any plans to extend it to windows node? One option maybe windows nodes can utilize this for enabling/disabling PD?

Reply:
I am closing this PR for now. But we will keep the option open to other meaningful short name.